src/OrchardCore.Modules/OrchardCore.Users/Services/CustomUserSettingsService.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.Users.Models;
+using YesSql;
+
+namespace OrchardCore.Users.Services;
+
+public class CustomUserSettingsService
+{
+    private readonly IContentManager _contentManager;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IAuthorizationService _authorizationService;
+    private readonly IContentDefinitionManager _contentDefinitionManager;
+    private readonly Lazy<IDictionary<string, ContentTypeDefinition>> _settingsTypes;
+    private readonly YesSql.ISession _session;
+
+    public CustomUserSettingsService(
+        IContentManager contentManager,
+        IHttpContextAccessor httpContextAccessor,
+        IAuthorizationService authorizationService,
+        IContentDefinitionManager contentDefinitionManager,
+        YesSql.ISession session)
+    {
+        _contentManager = contentManager;
+        _httpContextAccessor = httpContextAccessor;
+        _authorizationService = authorizationService;
+        _contentDefinitionManager = contentDefinitionManager;
+        _settingsTypes = new Lazy<IDictionary<string, ContentTypeDefinition>>(
+            () => _contentDefinitionManager
+                    .ListTypeDefinitions()
+                    .Where(x => x.GetStereotype() == "CustomUserSettings")
+                    .ToDictionary(x => x.Name));
+        _session = session;
+    }
+
+    public IEnumerable<string> GetAllSettingsTypeNames()
+    {
+        return _settingsTypes.Value.Keys;
+    }
+
+    public IEnumerable<ContentTypeDefinition> GetAllSettingsTypes()
+    {
+        return _settingsTypes.Value.Values;
+    }
+
+    public IEnumerable<ContentTypeDefinition> GetSettingsTypes(params string[] settingsTypeNames)
+    {
+        foreach (var settingsTypeName in settingsTypeNames)
+        {
+            if (_settingsTypes.Value.TryGetValue(settingsTypeName, out ContentTypeDefinition settingsType))
+            {
+                yield return settingsType;
+            }
+        }
+    }
+
+    public ContentTypeDefinition GetSettingsType(string settingsTypeName)
+    {
+        _settingsTypes.Value.TryGetValue(settingsTypeName, out ContentTypeDefinition settingsType);
+        return settingsType;
+    }
+
+    public async Task<bool> CanUserCreateSettingsAsync(ContentTypeDefinition settingsType)
```

Comment:
This method is not needed. Instead use the _authorizationService in the consumer class.